tty/pc-tty/ttypc_kbd.c

```diff
@@ -402,165 +417,205 @@ static char *_ttypc_kbd_get(ttypc_t *ttypc)
 
 
 /* Keyboard interrupt handler */
-static int ttypc_kbd_interrupt(unsigned int n, void *arg)
+static int _ttypc_kbd_interrupt(unsigned int n, void *arg)
 {
-	ttypc_t *ttypc = (ttypc_t *)arg;
-
-	return ttypc->kcond;
+	return 0;
 }
 
 
-static void ttypc_kbd_ctlthr(void *arg)
+/* Macros for distinguishing the origin of pending output */
+#define OUTPUT_PENDING(status)       (((status) & (1u << 0u)) != 0)
+#define KBD_OUTPUT_PENDING(status)   (((status) & (1u << 5u)) == 0)
+#define MOUSE_OUTPUT_PENDING(status) (((status) & (1u << 5u)) != 0)
+
+
+static void ttypc_kbd_handle_event(ttypc_t *ttypc, unsigned char b)
 {
-	ttypc_t *ttypc = (ttypc_t *)arg;
 	ttypc_vt_t *cvt;
 	char *s, k;
 	char buff[10];
 	unsigned char m;
 
-	mutexLock(ttypc->klock);
-	for (;;) {
-		/* Wait for character codes to show up in keyboard output buffer */
-		while (!(inb((void *)((uintptr_t)ttypc->kbd + 4)) & 0x01))
-			condWait(ttypc->kcond, ttypc->klock, 0);
-
-		mutexLock(ttypc->lock);
-		mutexLock((cvt = ttypc->vt)->lock);
+	mutexLock(ttypc->lock);
+	mutexLock((cvt = ttypc->vt)->lock);
```

Comment:
`cvt = ttypc->vt` - side effects inside function params, as in the other comments